src/backend/access/aocs/aocssegfiles.c

```diff
@@ -454,6 +454,136 @@ GetAOCSSSegFilesTotals(Relation parentrel, Snapshot appendOnlyMetaDataSnapshot)
 	return totals;
 }
 
+/*
+ * GetAOCSSSegFilesTotals
+ *
+ * Get the total FileSegTotals information for a specific AO table
+ * from the pg_aoseg tables on the entire cluster.
+ * Note: result is palloced, caller should free it.
+ */
+FileSegTotals *GetAOCSSegFilesTotalsCluster(Relation parentrel)
+{
+	FileSegTotals *result;
+	StringInfoData sqlstmt;
+	Relation	aosegrel;
+	volatile bool connected = false;
+	Oid			segrelid = InvalidOid;
+
+	Assert(Gp_role == GP_ROLE_DISPATCH);
+	Assert(RelationIsAoCols(parentrel));
+
+	GetAppendOnlyEntryAuxOids(RelationGetRelid(parentrel), NULL,
+							  &segrelid, NULL, NULL, NULL, NULL);
+	Assert(OidIsValid(segrelid));
+
+	result = (FileSegTotals *) palloc0(sizeof(FileSegTotals));
+
+	/*
+	 * open the aoseg relation
+	 */
+	aosegrel = heap_open(segrelid, AccessShareLock);
+
+	initStringInfo(&sqlstmt);
+	appendStringInfo(&sqlstmt, "select tupcount, varblockcount, vpinfo "
+					 "from gp_dist_random('%s.%s')",
+					 get_namespace_name(RelationGetNamespace(aosegrel)),
+					 RelationGetRelationName(aosegrel));
+
+	heap_close(aosegrel, AccessShareLock);
+
+	/*
+	 * Temporarily disable ORCA because it's slow to start up, and it
+	 * wouldn't come up with any better plan for the simple queries that
+	 * we run.
```

Comment:
we can write the comment in two lines not exceeding 80 symbols border. There are ORCA and Orca in one comment. I prefer Orca, because the optimizer name is not an abbreviation
```suggestion
	 * Temporarily disable Orca because it's slow to start up, and it wouldn't
	 * come up with any better plan for the simple queries that we run.
```

Reply:
Updated